p2p/simulations/network.go

```diff
@@ -151,11 +151,30 @@ func (self *Node) String() string {
 }
 
 func (self *Node) NodeInfo() *p2p.NodeInfo {
-	info := self.Node.NodeInfo()
+  var info *p2p.NodeInfo
+  if self.Node != nil {
+	  info = self.Node.NodeInfo()
+  } else {
+    info = &p2p.NodeInfo{}
+  }
 	info.Name = self.Config.Name
 	return info
 }
 
+func (self *Node) MarshalJSON() ([]byte, error) {
```

Comment:
Add a comment explaining why it is necessary to explicitly implement the `json.Marshaler` interface.